### PR TITLE
[ADD] business_requirement_deliverable_project_task_mandatory

### PR DESCRIPTION
**Pull request by victor-ditay**

### CODE REVIEW

**Comment by victor-ditay** [comment]
@Elico-Corp/odoo-reviewers Please review and note in the PR name Task and BR

**Comment by victor-ditay** [comment]
Better work with updated version of:
OCA/project 8.0 regards PRs  https://github.com/OCA/project/pull/239 and https://github.com/OCA/project/pull/240

Need to have merge on `@noahzaozao BR repository` : PR https://github.com/OCA/business-requirement/pull/89

cc @Elico-Corp/odoo-reviewers

File: business_requirement_deliverable_project_task_mandatory/README.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======================================================
+Business Requirement Deliverable Project Task Mandatory
+=======================================================
+
+This module extends the functionality of Project Category
+Business Requirement Project and allow to set a Category
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/README.rst]: allows

File: business_requirement_deliverable_project_task_mandatory/README.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+that will make required have all task linked to a Business Requirement.
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/README.rst]: that will make all tasks linked to a Business Requirement required.

File: business_requirement_deliverable_project_task_mandatory/README.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to Project Category
+#. Select one Category and check br_required
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/README.rst]: "BR Required"

File: business_requirement_deliverable_project_task_mandatory/README.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+Credits
+=======
+
+Contributors
+------------
+
+* Eric Caudal <eric.caudal@elico-corp.com>
+* Victor Martin <victor.martin@elico-corp.com>
+
+Maintainer
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/README.rst:38]: I think this module could be added to the OCA

File: business_requirement_deliverable_project_task_mandatory/models/project_project_category.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+from openerp import fields, models
+
+
+class ProjectProjectCategory(models.Model):
+    _inherit = 'project.project.category'
+
+    br_required = fields.Boolean(
+        'Business Requirement required',
+        help='If marked as true the task linked to the project'
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/models/project_project_category.py]: If marked as true the business requirement link in the task linked to this project will be mandatory

File: business_requirement_deliverable_project_task_mandatory/models/project_task.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# import json
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/models/project_task.py]: remove dead code

File: business_requirement_deliverable_project_task_mandatory/models/project_task.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/models/project_task.py]: be consistent with the copyright

File: business_requirement_deliverable_project_task_mandatory/models/project_task.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+    @api.constrains(
+        'project_id',
+        'business_requirement_id',
+        'br_required')
+    def _check_br_required(self):
+        for t in self:
+            if t.br_required and not t.business_requirement_id.id:
+                raise models.ValidationError(
+                    'Project related category requires'
```
Comment (elicoidal) [inline business_requirement_deliverable_project_task_mandatory/models/project_task.py]: The Business Requirement is mandatory for this type of project category

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details, mostly English.
I would propose it to the OCA as well
About the name, what about business_requirement_task_mandatory

**Comment by victor-ditay** [comment]
> some details, mostly English.
I would propose it to the OCA as well
About the name, what about business_requirement_task_mandatory

@elicoidal I think the name should reflect at least one the modules that depends on. So if check depends will see this name of module but I could change it if you want

File: business_requirement_deliverable_project_task_mandatory/README.rst (hunk 4 above)
Comment (martintorresvm) [reply business_requirement_deliverable_project_task_mandatory/README.rst:38]: Better later?
Actually our module for Project Category works fine and I rather avoid new migration.

The dependency is solved and is just a matter to improve later to converge with OCA/project modules and deprecate our `project_project_category`

**Comment by elicoidal** [comment]
> I think the name should reflect at least one the modules that depends on. So if check depends will see this name of module but I could change it if you want

Agree: dont change the name.

File: business_requirement_deliverable_project_task_mandatory/README.rst (hunk 4 above)
Comment (elicoidal) [reply business_requirement_deliverable_project_task_mandatory/README.rst:38]: Actually the project category doesnot work fine AFAIK and we should switch to the new one asap

**Review by elicoidal — APPROVED** [review]

**Comment by elicoidal** [comment]
@noahzaozao @seb-elico @xuan139 please review

**Comment by victor-ditay** [comment]
> Better work with updated version of:
OCA/project 8.0 regards PRs OCA/project#239 and OCA/project#240
Need to have merge on @noahzaozao BR repository : PR OCA/business-requirement#89

All these PR are now merged. Could be merged this PR @Elico-Corp/odoo-reviewers?

**Pull request merged**